### ISSUE TITLE
unused-imports uses hardcoded list for typing.TYPE_CHECKING

### ISSUE THREAD

**Issue by dcbaker** [PyCQA/pylint#3846]
In meson we import the `typing` module as `T`, as a sort of compromise between using the full `typing` name, and polluting the local namespace.

### Steps to reproduce
```python
import typing as T
if T.TYPE_CHECKING:
    import Foo
```

Looking at the code it uses a hardcoded set `{'typing.TYPE_CHECKING', 'TYPE_CHECKING'}`, using any other value will be ignored and spurious errors will ensue.

For my use, being able to add T.TYPE_CHECKING to the pylintrc would be sufficient, but maybe this can resolved in a more generic way?

**Comment by graingert** [comment]
pylint should definitely track what `typing` is imported as, and really must not trackin

but you'll probably not need this as most of the common stuff in `typing` is now moved to `builtins` and `collections.abc`

Then all the uncomon stuff `typing.Protocol`, `typing.TypeVar`, `typing.cast` justify using the full `import typing` name

**Comment by graingert** [comment]
there's also 
```python
MYPY = False
if MYPY:
    import bar
```

to consider

**Comment by dcbaker** [comment]
We need to support all versions of python the python foundation does. We're just getting ready to drop 3.5 support after our next release. So using the not in typing names won't happen for a lot of years for us.

**Comment by graingert** [comment]
@dcbaker you won't need it (in forward annotations) as soon as mypy supports pep 585

**Comment by PCManticore** [comment]
Thanks for the report; I agree that we need to improve the situation here.
